### ISSUE TITLE
Bug: `renderToString` always generates an empty html comment

### ISSUE THREAD

**Issue by leifmarcus** [vuejs/test-utils#2448]
**Describe the bug**
When using `await renderToString(MyComponent)`, the return value is always `<---->` instead of the component markup

**To Reproduce**
Please open an [example implementation on Stackblitz](https://stackblitz.com/edit/github-rm9s3m-rsklhr?file=src%2Fcomponents%2F__tests__%2FHelloWorld.spec.ts)

**Expected behavior**
The markup should be rendered as expected when rendering the SSR string.

**Related information:**
```
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  npmPackages:
    @vue/test-utils: ^2.4.6 => 2.4.6 
    vitest: ^1.6.0 => 1.6.0 
    vue: 3.3.13 => 3.3.13 
```

**Additional context**
What I think is wired is, that this was working a few days ago. I'm not sure, what did change so that the rendering does not work anymore in the `vitest` context.

**Comment by cexbrayat** [comment]
Hi @leifmarcus 

I see you're using Vue v3.3. Can you try with the latest Vue release to see if it solves your issue, or provide additional context on the error?

**Comment by leifmarcus** [comment]
@cexbrayat, Thanks for responding. Sadly I need to use the latest `3.3.` Vue version, as the new `3.4` is currently not working with other tools. Nevertheless, I found, that when using Single File Components, it works as expected. It seems to be a problem with Components defined inside the test file.

**Comment by cexbrayat** [comment]
If I bump your repro to Vue v3.4, it looks like it works as expected no? So this is probably an issue with Vue v3.3 itself, that has been solved in v3.4. In that case, this is not something we can help with in VTU. Your tests should pass when you'll be able to upgrade
